src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Controls/PopupControlService.cs

```diff
@@ -1558,39 +1558,56 @@ private void AddPoints(PointList points, in NativeMethods.RECT rect, bool rectIs
+                System.Windows.Point pt = Mouse.PrimaryDevice.NonRelativePosition;
+
+                // translate the point to our source's coordinates, if necessary
+                // (e.g. if the tooltip's owner comes from a window with capture,
+                // such as the popup of a ComboBox)
+                if (mouseSource != _source)
                 {
-                    // get the coordinates of the current mouse point, relative to our PresentationSource
-                    System.Windows.Point pt = Mouse.PrimaryDevice.GetPosition(rootElement);
-
-                    // check whether the point lies within the hull
-                    return ContainsPoint(_source, (int)pt.X, (int)pt.Y);
-
-                    // NOTE: GetPosition doesn't actually return the position of the current mouse point,
-                    // but rather the last recorded position.  (See MouseDevice.GetScreenPositionFromSystem,
-                    // which says that "Win32 has issues reliably returning where the mouse is".)
-                    // This causes a small problem when (a) the PresentationSource has capture, e.g.
-                    // the popup of a ComboBox, and (b) the mouse moves to a position that lies outside both the
-                    // capturing PresentationSource (popup window) and the input-providing PresentationSource
-                    // (main window).  The MouseDevice only records positions within the input-providing
-                    // PresentationSource, so we'll test the position where the mouse left the main window,
-                    // rather than the current position.
-                    //      This means we may leave a tooltip open even when the mouse leaves its SafeArea,
-                    // but only when the tooltip belongs to a capturing source, and the "leaving the SafeArea"
-                    // action occurs outside the surrounding main window.  For our example, it can happen
-                    // when the ComboBox is close to the edge of the main window so that a tooltip from its
-                    // popup content extends beyond the main window.
-                    //      This can only be fixed by changing MouseDevice.GetScreenPositionFromSystem to
-                    // use a "better way" to find the current mouse position, which allegedly needs work from the OS.
-                    // But we can live with this behavior, because
-                    //  * this is a corner case - tooltips from popup content that extend beyond the main window
-                    //  * the effect is transient - the tooltip will close when the user dismisses the popup
-                    //  * there's no accessibility issue - WCAG 2.1 only requires that the tooltip stays open under
-                    //      proscribed conditions, not that it has to close when the conditions cease to apply
+                    System.Windows.Point ptScreen = PointUtil.ClientToScreen(pt, mouseSource);
```

Comment:
you may want to use direct mapping if it helps perf: ref:https://github.com/dotnet/winforms/blob/8b099dc847ccc153782b7c1591f677ff779ae9a8/src/System.Windows.Forms.Primitives/src/Interop/User32/Interop.MapWindowPoints.cs